src/streamlink/plugins/bbciplayer.py

```diff
@@ -87,26 +93,36 @@ def _extract_nonce(cls, http_result):
         # Return the nonce we can use for future queries
         return goto_url_query['nonce']
 
-    def _validate_login(self, result_history, ptrt_url):
+    def _validate_login(self, result, ptrt_url):
         """
         Check that the last redirect in the auth flow, matches our provided snapback URL
 
-        :param result_history:
+        :param result: HTTP response from the BBC ID sign-in endpoint
         :param ptrt_url: The snapback URL to redirect to after successful authentication
-        :type result_history: list[requests.Response]
+        :type result: requests.Response
         :type ptrt_url: string
         :return: Whether authentication was successful
         :rtype: bool
         """
-        no_history = len(result_history) == 0
+        no_history = len(result.history) == 0
         bad_goto_url = True
         if not no_history:
-            goto_url = str(result_history[-1].url)
+            goto_url = str(result.url)
             # If the user did not provide a 'www.' URL, we have to remove it so string comparison works.
-            prefix = "http://www."
-            if goto_url.startswith(prefix) and not ptrt_url.startswith(prefix):
-                goto_url = "http://" + goto_url[len(prefix):]
+
+            goto = self.www_sub_domain_re.match(goto_url).groupdict()
+            ptrt = self.www_sub_domain_re.match(ptrt_url).groupdict()
+
+            if goto['has_www'] != ptrt['has_www']:
+                # It's not a typo to use "goto" as the replacement in both.
+                # The intention here is to check whether the ptrt_url will result in the same
+                # resource being returned to the user, so for the purposes of validation,
+                # we can ignore both the protocol, and any www subdomain.
+                goto_url = self.www_sub_domain_re.sub(goto['prefix'] + "://", goto_url)
+                ptrt_url = self.www_sub_domain_re.sub(goto['prefix'] + '://', ptrt_url)
```

Comment:
still not fully working,


```
streamlink http://www.bbc.co.uk/iplayer/live/cbeebies best -l debug

 ----------- goto_url - if not no_history
https://www.bbc.co.uk/iplayer/live/cbeebies
 ----------- ptrt url
http://www.bbc.co.uk/iplayer/live/cbeebies
goto
{'prefix': 'https', 'has_www': 'www.'}
ptrt
{'prefix': 'http', 'has_www': 'www.'}
 ----------- goto_url after www
https://www.bbc.co.uk/iplayer/live/cbeebies
 ----------- ptrt url after www
http://www.bbc.co.uk/iplayer/live/cbeebies
[plugin.bbciplayer][error] Could not authenticate, could not find the authentication nonce
[plugin.bbciplayer][error] Could not authenticate, check your username and password
```

**but is this even needed ?**
if the **login fails** there is `no_history` its just **one url**

`https://account.bbc.com/signin?ptrt=http%3A%2F%2Fwww.bbc.co...`

if the **login** is **successful** there are like **3 - 4 urls** in the history

just checking for the `history == 0` should be enough

there is no cache in this plugin, so it would be always like that.

is there something that would not work, with only checking the history?

Reply:
Now that you mention it, there's I can not think of anything. That said, I don't have insight to the BBC ID api, so I don't know what the possible scope of returns are. 

I think it's just a carry over from the previous. Unless you have any insight into why it's there, I guess I should probably remove it.

Reply:
well you added a history check,
i think that would be enough, since a failed login doesn't add history.